resources/translations.ts

```diff
@@ -0,0 +1,150 @@
+    fr: 'Vous commencez une synthèse d\'essai pour une? \ue0bb(?<recipe>.*)\\.',
+    ja: '(?<player>\\y{Name})は\ue0bb(?<recipe>.*)の製作練習を開始した。',
+    cn: '(?<player>\\y{Name})开始练习制作\ue0bb(?<recipe>.*)。',
+    ko: '\ue0bb(?<recipe>.*) 제작 연습을 시작합니다\\.',
+  },
+  craftingFinish: {
+    en: 'You synthesize (?<count>(an?|\\d+) )?\ue0bb(?<recipe>.*)(\ue03c)?\\.',
+    de: 'Du hast erfolgreich (?<count>(ein(e|es|em|er)?|\\d+) )?(?<recipe>.*)(\ue03c)? hergestellt\\.',
+    fr: 'Vous fabriquez (?<count>(une?|\\d+) )?\ue0bb(?<recipe>.*)(\ue03c)?\\.',
+    ja: '(?<player>\\y{Name})は\ue0bb(?<recipe>.*)(\ue03c)?(×(?<count>\\d+))?を完成させた！',
+    cn: '(?<player>\\y{Name})制作“\ue0bb(?<recipe>.*)(\ue03c)?”(×(?<count>\\d+))?成功！',
+    ko: '(?<player>\\y{Name}) 님이 \ue0bb(?<recipe>.*)(\ue03c)?(×(?<count>\\d+)개)?(을|를) 완성했습니다!',
+  },
+  trialCraftingFinish: {
+    en: 'Your trial synthesis of \ue0bb(?<recipe>.*) proved a success!',
+    de: 'Die Testsynthese von \ue0bb(?<recipe>.*) war erfolgreich!',
+    fr: 'Votre synthèse d\'essai pour fabriquer \ue0bb(?<recipe>.*) a été couronnée de succès!',
+    ja: '(?<player>\\y{Name})は\ue0bb(?<recipe>.*)の製作練習に成功した！',
+    cn: '(?<player>\\y{Name})练习制作\ue0bb(?<recipe>.*)成功了！',
+    ko: '\ue0bb(?<recipe>.*) 제작 연습에 성공했습니다!',
+  },
+  craftingFail: {
+    en: 'Your synthesis fails!',
+    de: 'Deine Synthese ist fehlgeschlagen!',
+    fr: 'La synthèse échoue\\.{3}',
+    ja: '(?<player>\\y{Name})は製作に失敗した……',
+    cn: '(?<player>\\y{Name})制作失败了……',
+    ko: '제작에 실패했습니다……\\.',
+  },
+  trialCraftingFail: {
+    en: 'Your trial synthesis of \ue0bb(?<recipe>.*) failed\\.{3}',
+    de: 'Die Testsynthese von \ue0bb(?<recipe>.*) ist fehlgeschlagen\\.{3}',
+    fr: 'Votre synthèse d\'essai pour fabriquer \ue0bb(?<recipe>.*) s\'est soldée par un échec\\.{3}',
+    ja: '(?<player>\\y{Name})は\ue0bb(?<recipe>.*)の製作練習に失敗した……',
+    cn: '(?<player>\\y{Name})练习制作\ue0bb(?<recipe>.*)失败了……',
+    ko: '\ue0bb(?<recipe>.*) 제작 연습에 실패했습니다……\\.',
+  },
+  craftingCancel: {
+    en: 'You cancel the synthesis\\.',
+    de: 'Du hast die Synthese abgebrochen\\.',
+    fr: 'La synthèse est annulée\\.',
+    ja: '(?<player>\\y{Name})は製作を中止した。',
+    cn: '(?<player>\\y{Name})中止了制作作业。',
+    ko: '제작을 중지했습니다\\.',
+  },
+  trialCraftingCancel: {
+    en: 'You abandoned trial synthesis\\.',
+    de: 'Testsynthese abgebrochen\\.',
+    fr: 'Vous avez interrompu la synthèse d\'essai\\.',
+    ja: '(?<player>\\y{Name})は製作練習を中止した。',
+    cn: '(?<player>\\y{Name})停止了练习。',
+    ko: '제작 연습을 중지했습니다\\.',
+  },
+};
+
+type LocaleRegexesObj = Record<keyof typeof localeLines, Record<Lang, RegExp>>;
+
+// Here overrides the build-in Proxy type definition
+// because it assumes that the source object type should
+// be the same as the target object, but here we use
+// proxy to dynamically return RegExp instead of strings
+// in the source object. So create a hacky Type to override
+// the default definition to make this work.
+interface OverrideProxyConstructor {
```

Comment:
Is it possible to not use a proxy here?  I recently sped up unittests a bunch by remove `Regexes.parse` lines (which these helpers both do).  One thing that was nice about the previous code is that it generates regexes for all languages once, and then never again.  I would prefer not to have the proxy dynamically compute it every time.  It's also a little bit suspicious to my eyes to have a Proxy for a Record (which should never return undefined) have a bunch of "return null" cases.

Is it possible to make a more generic object and then use `as LocaleRegexesObj` or something to tell typescript that it's correct? (I tried doing that, but then it thought that LocaleRegex.countdownCancel might be undefined, even with the `as`? Maybe you can figure out something else?) Alternatively, is there some other way of constructing this?

Reply:
Correct, I tried before but I found it is hard to simply add types on the original code.
I think if not to use proxy, the better way is to create a dummy class which caches regexes at the runtime and export the entries to access to `NetRegex` and `Regex`?

Reply:
Could the proxy precalculate them all ahead of time and access the precalculated object that used to be returned?